java-frontend/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -254,6 +255,10 @@ private void prune() {
         block.id = id;
         id += 1;
       }
+      inactiveBlocks.removeAll(blocks);
```

Comment:
This part has nothing to do with the current ticket. Therefore it should not be included in this commit. A dedicated test case is also required.